### PR TITLE
Refactor rhel-jboss-templates

### PR DESCRIPTION
**Pull request by backwind1233**
## Changes in this PR
1. Refactor: remove Git refJavaee in JBoss Offers
2. Refactor: refactor bicep scripts to use version placeholders so all the versions can be managed in only one [file](https://github.com/Azure/rhel-jboss-templates/blob/a57552df4c8c6bc9a238361bc0bb41b3d5c8e645/utilities/azure-common.properties).
3. Refactor the project's maven structure, use the root [pom.xml](https://github.com/Azure/rhel-jboss-templates/blob/a57552df4c8c6bc9a238361bc0bb41b3d5c8e645/pom.xml) to manage all the versions and configurations.
4. Refactor all the workflows
5. Fix all errors in the workflows.

### Workflow test results
| Workflow                         | Status | Link                                                         |
| -------------------------------- | ------ | ------------------------------------------------------------ |
| build-artifact-byos-multivm.yaml | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7383867539) |
| build-artifact-byos-single.yaml  | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7383868214) |
| build-artifact-byos-vmss.yaml    | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7383172867) |
| build-artifact-eap-aro.yaml      | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7383172335) |
| build-artifact-payg-multivm.yaml | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7383225068) |
| build-artifact-payg-single.yaml  | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7383171437) |
| build-artifact-payg-vmss.yaml    | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7383171237) |
| validate-byos-multivm.yaml       | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7385249431) |
| validate-byos-singlenode.yaml    | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7385200701) |
| validate-byos-vmss.yaml          | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7385762328) |
| validate-payg-multivm.yaml       | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7392860460) |
| validate-payg-singlenode.yaml    | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7394523099) |
| validate-payg-vmss.yaml          | ✅      | [link](https://github.com/azure-javaee/rhel-jboss-templates/actions/runs/7393982468) |

### byos-multivm ✅   8/8
image

### byos-vmss ✅ 4/4
image

### byos-single ✅   4/4
image

### payg-vmss ✅  4/4
image

### payg-single ✅ 4/4

image

### payg-multivm ✅ 6/6
image

### CODE REVIEW

File: eap74-rhel8-payg/src/main/bicep/mainTemplate.bicep [hunk 1]
```diff
@@ -321,7 +321,7 @@ module dbConnectionStartPid './modules/_pids/_pid.bicep' = if (enableDB) {
   ]
 }
 
-resource vmName_jbosseap_setup_extension 'Microsoft.Compute/virtualMachines/extensions@2022-08-01' = {
+resource vmName_jbosseap_setup_extension 'Microsoft.Compute/virtualMachines/extensions@${azure.apiVersionForVirtualMachineScaleSets}' = {
```
Comment (backwind1233) [inline eap74-rhel8-payg/src/main/bicep/mainTemplate.bicep:324]: fix this, it should be apiVersionForVirtualMachineExtensions

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    <pluginRepositories>
+        <pluginRepository>
+            <id>github</id>
+            <name>GitHub Packages</name>
+            <url>https://maven.pkg.github.com/azure-javaee/azure-javaee-iaas</url>
+        </pluginRepository>
+    </pluginRepositories>
+
+    <properties>
+        <template.validation.tests.directory>${basedir}/../arm-ttk/arm-ttk</template.validation.tests.directory>
```
Comment (backwind1233) [inline pom.xml:37]: add some comments here

**Review by edburns — APPROVED** [review]

**Review by edburns — APPROVED** [review]
LGTM

**Review by edburns — APPROVED** [review]
LGTM

**Pull request merged**